zcash_client_sqlite/src/wallet/init/migrations/shardtree_support.rs

```diff
@@ -180,10 +204,44 @@ impl RusqliteMigration for Migration {
                     updated_note_positions.insert(witnessed_position);
                 }
 
-                shard_tree.insert_witness_nodes(witness, BlockHeight::from(block_height))?;
+                let _ = shard_tree.insert_witness_nodes(witness, BlockHeight::from(block_height));
```

Comment:
Even discard `ShardTreeError::Storage` here? That seems wrong to me. Shouldn't we remember the error and reraise it below somewhere?

Reply:
`<CachingShardStore<_> as ShardStore>::Error = Infallible`, because the `ShardStore` trait impl operates purely in memory. So we cannot raise this as the parent error type has no case for `Infallible`.

Reply:
Oh wait, what we need to do here is the opposite of what you suggest: we need to discard _only_ `ShardTreeError::Storage`, as the others _can_ be mapped up.